web/packages/shared/components/TextEditor/TextEditor.jsx

```diff
@@ -104,20 +111,59 @@ class TextEditor extends React.Component {
 
   render() {
     const { bg = 'levels.sunken' } = this.props;
+    const hasButton = this.props.copyButton || this.props.downloadButton;
+
     return (
       <StyledTextEditor bg={bg}>
         <div ref={e => (this.ace_viewer = e)} />
+        {hasButton && (
+          <ButtonSection>
+            {this.props.copyButton && (
+              <EditorButton
+                test-id
+                title="Copy to clipboard"
+                onClick={() => copyToClipboard(this.editor.session.getValue())}
+              >
+                <Copy size="18" />
+              </EditorButton>
+            )}
+            {this.props.downloadButton && (
+              <EditorButton
+                title="Download"
+                onClick={() =>
+                  downloadObject(
+                    this.props.downloadFileName,
+                    this.editor.session.getValue()
+                  )
+                }
+              >
+                <Download size="18" />
+              </EditorButton>
+            )}
+          </ButtonSection>
+        )}
       </StyledTextEditor>
     );
   }
 }
 
-export default TextEditor;
-
 function getMode(docType) {
   if (docType === 'json') {
     return 'ace/mode/json';
   }
 
   return 'ace/mode/yaml';
 }
+
+const EditorButton = styled(ButtonSecondary)`
+  padding: 12px;
+  background-color: transparent;
+`;
+
+const ButtonSection = styled(Flex)`
+  position: absolute;
```

Comment:
This is a risky, it can cover text:
<img width="656" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/178f0b9f-3402-4f82-8a15-451d2540c8ee">

Perhaps it would be better to do the same as in the terminal?
<img width="269" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/7352a1c4-3c58-4c18-8cf7-cf838ee6619a">